jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java

```diff
@@ -533,11 +543,86 @@ public void abortWorkItem(WorkItem workItem,
     }
 
     private ClassLoader getInternalClassLoader() {
-        if (this.classLoader != null) {
-            return this.classLoader;
+        ClassLoader cl = this.classLoader != null ? classLoader : Thread.currentThread().getContextClassLoader(), parent = cl;
+        Collection<File> uris = new HashSet<>();
+        do {
+            if (parent.getClass().getSimpleName().equals("ModuleClassLoader")) {
+                try {
+                    getJarsFromModuleClassLoader(parent, uris);
+                } catch (ReflectiveOperationException e) {
+                    throw new IllegalStateException("Problem calculating classpath", e);
+                }
+            }
+            parent = parent.getParent();
+        } while (parent != null);
+        if (!uris.isEmpty()) {
+            cl = new CustomClassLoader(uris, cl);
+        }
+        return cl;
+    }
+
+    private static class CustomClassLoader extends URLClassLoader {
+
+        private URL[] hackUrls;
```

Comment:
bad naming. this is a classpath phisical location. hackUrls says very little. 
urlLocationClassPathLibs or so

Reply:
yes, I changed to jarUrls, which is shorter that urlLocationClasspathLibs and more accurate than hackURLs